### ISSUE TITLE
Conflicting definitions when they are identical

### ISSUE THREAD

**Issue by arienkock** [alanctgardner/gogen-avro#34]
My use case is that I have IDL protocols generating multiple avsc files which overlap (default behavior for `avro-tools-1.8.2.jar idl2schemata`). When registering definitions it would be nice if the contents was checked for equality to determine if it's ok to proceed or not. Right now, any definition with duplicate FQN or aliases will abort the processing.

Relevant parts:
https://github.com/alanctgardner/gogen-avro/blob/master/types/schema.go#L71
https://github.com/alanctgardner/gogen-avro/blob/master/types/schema.go#L77

**Comment by panxh** [comment]
I have the same case, compile avsc files one by one will work out. however, my problem is gogen-avro will generate an identical file named "primitive.go" for every avsc file, thus the prev one will be replaced by  the later. Any idea?

**Comment by alanctgardner** [comment]
@panxh @arienkock This issue should be fixed in master and in the new release 5.0.3, so you can now compile all your files at once without errors about the duplicate definitions. Thanks for your patience!

**Comment by tzachshabtay** [comment]
@actgardner this doesn't seem to work for enums in a different file but same namespace.

To reproduce:

file1.idl:
```
@namespace("myname")
protocol MyProtocol {
    enum type {
        type1,
        type2
    }
}
```

file2.idl
```
@namespace("myname")
protocol MyProtocol2 {
  record MyRecord {
      type mytype;
  }
}
```

For this scenario, IDL generates two avsc files, in the type.avsc file there's a "namespace: myname" attribute associated with type, but in the myrecord.avsc file there's no namespace attribute associated with type, so the `deepequal` check you added returns false and the error is still generated.
